### PR TITLE
[ready for review] Error wrapper

### PR DESCRIPTION
**Pull request by zhenfeizhang**
API-breaking change
need to batch-merge multiple PRs

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -27,12 +26,10 @@ ark-serialize = { version = "0.2.0", default-features = false, features = [ "der
 blake2 = { version = "0.9.1", default-features = false }
 
 # Manta dependencies
-manta-crypto = { branch = "master", git = "https://github.com/Manta-Network/manta-crypto/", default-features = false }
+manta-crypto = { branch = "error-wrapper", git = "https://github.com/Manta-Network/manta-crypto/", default-features = false }
```
Comment (zhenfeizhang) [inline Cargo.toml:29]: will update the branch once merged

**Review by ghzlatarev — APPROVED** [review]

File: src/serdes.rs [hunk 2]
```diff
@@ -18,181 +18,182 @@ use super::*;
...
-		asset.priv_info = MantaAssetPrivInfo::deserialize(&mut reader);
+		reader.read_exact(&mut asset.commitment)?;
+		reader.read_exact(&mut asset.void_number)?;
+		asset.pub_info = MantaAssetPubInfo::deserialize(&mut reader)?;
+		asset.priv_info = MantaAssetPrivInfo::deserialize(&mut reader)?;
 
-		let commit_param = CommitmentParam::deserialize(COMMIT_PARAM.data);
-		assert!(asset.sanity(&commit_param));
+		let commit_param = CommitmentParam::deserialize(COMMIT_PARAM.data)?;
+		assert!(asset.sanity(&commit_param)?);
```
Comment (gabcbrown) [inline src/serdes.rs:47]: Do we want this to cause a panic if the check fails? Or can this be another error type that we handle in a more graceful way?

File: src/serdes.rs [hunk 3]
```diff
@@ -18,181 +18,182 @@ use super::*;
...
-			prepared: MantaAssetPreparedReceiver::deserialize(&mut reader),
-			spend: MantaAssetReceiverSpendingInfo::deserialize(&mut reader),
+			prepared: MantaAssetPreparedReceiver::deserialize(&mut reader)?,
+			spend: MantaAssetReceiverSpendingInfo::deserialize(&mut reader)?,
 		};
 
-		let commit_param = CommitmentParam::deserialize(COMMIT_PARAM.data);
-		assert!(receiver.sanity(&commit_param));
+		let commit_param = CommitmentParam::deserialize(COMMIT_PARAM.data)?;
+		assert!(receiver.sanity(&commit_param)?);
```
Comment (gabcbrown) [inline src/serdes.rs:112]: Same here as previous comment.

File: src/serdes.rs (hunk 2 above)
Comment (zhenfeizhang) [reply src/serdes.rs:47]: good catch. we don't want panic. A new error type is in order

File: Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "manta-asset"
-version = "3.0.0"
+version = "0.1.0"
 authors = ["Manta Network <contact@manta.network>"]
 edition = "2018"
 license = "Apache-2.0"
```
Comment (stechu) [inline Cargo.toml:6]: Please change it to GPL V3, be consistent with the licence of the repo.

**Review by stechu — CHANGES_REQUESTED** [review]

**Pull request merged**